src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -695,6 +1056,39 @@ def get_synopsys_rm_tarball(self, product: str, settings_key: str = "") -> str:
         else:
             return synopsys_rm_tarball
 
+class MentorTool(HammerTool):
+    """ Mix-in trait with functions useful for Mentor-Graphics-based tools. """
+    @property
+    def env_vars(self) -> Dict[str, str]:
+        """
+        Get the list of environment variables required for this tool.
+        Note to subclasses: remember to include variables from super().env_vars!
+        """
+        result = dict(super().env_vars)
+        result.update({
+            "MGLS_LICENSE_FILE": self.get_setting("mentor.MGLS_LICENSE_FILE")
+        })
+        return result
+
+    def version_number(self, version: str) -> int:
+        """
+        Assumes versions look like NAME-YYYY.MM-SPMINOR.
+        Assumes less than 100 minor versions.
+        """
+        return 0 # TODO john
+
+class MentorCalibreTool(MentorTool):
+    """ Mix-in trait for Mentor's Calibre tool suite. """
+    @property
+    def env_vars(self) -> Dict[str, str]:
+        """
+        Get the list of environment variables required for this tool.
+        Note to subclasses: remember to include variables from super().env_vars!
+        """
+        # TODO MGC_HOME
```

Comment:
Unresolved TODO? Should this go in MentorTool?

Reply:
Yes, I will fix this.

Reply:
This actually needs to be set in the CAD plugin, since this is really just pointing to the Calibre version.